test/e2e/e2e_setup_test.go

```diff
@@ -0,0 +1,713 @@
+
+	bankGenState.DenomMetadata = append(bankGenState.DenomMetadata, banktypes.Metadata{
+		Description: "An example stable token",
+		Display:     photonDenom,
+		Base:        photonDenom,
+		Symbol:      photonDenom,
+		Name:        photonDenom,
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    photonDenom,
+				Exponent: 0,
+			},
+		},
+	})
+
+	bz, err = cdc.MarshalJSON(&bankGenState)
+	s.Require().NoError(err)
+	appGenState[banktypes.ModuleName] = bz
+
+	var genUtilGenState genutiltypes.GenesisState
+	s.Require().NoError(cdc.UnmarshalJSON(appGenState[genutiltypes.ModuleName], &genUtilGenState))
+
+	// generate genesis txs
+	genTxs := make([]json.RawMessage, len(s.chain.validators))
+	for i, val := range s.chain.validators {
+		createValmsg, err := val.buildCreateValidatorMsg(stakeAmountCoin)
+		s.Require().NoError(err)
+
+		signedTx, err := val.signMsg(createValmsg)
+		s.Require().NoError(err)
+
+		txRaw, err := cdc.MarshalJSON(signedTx)
+		s.Require().NoError(err)
+
+		genTxs[i] = txRaw
+	}
+
+	genUtilGenState.GenTxs = genTxs
+
+	bz, err = cdc.MarshalJSON(&genUtilGenState)
+	s.Require().NoError(err)
+	appGenState[genutiltypes.ModuleName] = bz
+
+	bz, err = json.MarshalIndent(appGenState, "", "  ")
+	s.Require().NoError(err)
+
+	genDoc.AppState = bz
+
+	bz, err = tmjson.MarshalIndent(genDoc, "", "  ")
+	s.Require().NoError(err)
+
+	// write the updated genesis file to each validator
+	for _, val := range s.chain.validators {
+		writeFile(filepath.Join(val.configDir(), "config", "genesis.json"), bz)
+	}
+}
+
+func (s *IntegrationTestSuite) initValidatorConfigs() {
+	for i, val := range s.chain.validators {
+		tmCfgPath := filepath.Join(val.configDir(), "config", "config.toml")
+
+		vpr := viper.New()
+		vpr.SetConfigFile(tmCfgPath)
+		s.Require().NoError(vpr.ReadInConfig())
+
+		valConfig := &tmconfig.Config{}
+		s.Require().NoError(vpr.Unmarshal(valConfig))
+
+		valConfig.P2P.ListenAddress = "tcp://0.0.0.0:26656"
+		valConfig.P2P.AddrBookStrict = false
+		valConfig.P2P.ExternalAddress = fmt.Sprintf("%s:%d", val.instanceName(), 26656)
+		valConfig.RPC.ListenAddress = "tcp://0.0.0.0:26657"
+		valConfig.StateSync.Enable = false
+		valConfig.LogLevel = "info"
+
+		var peers []string
+
+		for j := 0; j < len(s.chain.validators); j++ {
+			if i == j {
+				continue
+			}
+
+			peer := s.chain.validators[j]
+			peerID := fmt.Sprintf("%s@%s%d:26656", peer.nodeKey.ID(), peer.moniker, j)
+			peers = append(peers, peerID)
+		}
+
+		valConfig.P2P.PersistentPeers = strings.Join(peers, ",")
+
+		tmconfig.WriteConfigFile(tmCfgPath, valConfig)
+
+		// set application configuration
+		appCfgPath := filepath.Join(val.configDir(), "config", "app.toml")
+
+		appConfig := srvconfig.DefaultConfig()
+		appConfig.API.Enable = true
+		appConfig.MinGasPrices = fmt.Sprintf("%s%s", minGasPrice, photonDenom)
+
+		srvconfig.WriteConfigFile(appCfgPath, appConfig)
+	}
+}
+
+func (s *IntegrationTestSuite) runEthContainer() {
```

Comment:
this isn't a blocker, but in umee, I modified this so it _pulls_ the image instead of building it. Since the image should never really change for geth, it makes it slightly faster.

Reply:
Dude, I'm so sorry -- ignore me. I didn't realize we did this only for hermes and something else. This is good 👍